.github/workflows/ci.yml

```diff
@@ -18,12 +18,12 @@ jobs:
         with:
           node-version: '16'
       - uses: actions/cache@v2
-        id: npm-cache
         with:
-          path: '**/node_modules'
+          path: ~/.npm
           key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-}
       - run: npm ci
-        if: steps.npm-cache.outputs.cache-hit != 'true'
```

Comment:
It seems like you're copying the example, but wouldn't we want to keep this? Is there a reason to still run `npm ci` in the event of a cache hit?

Reply:
Yeah. We do need to `npm ci`. This approach caches the npm cache. This npm cache is not `node_modules`. It is another central location where downloads are kept. Those downloads do not include generated code or native deps downloaded outside of npm